### PR TITLE
DR-91 pdao - implement pdao for create study

### PR DESCRIPTION
**Pull request by ddietterich**
pDAO works for create study. In addition, I have added test categories so that the unit test for the pdao does not get run when the target is `test`. The test targets are now:
- test - runs Unit tests and StairwayUnit tests
- testConnected - runs the pDAO test
- testAll - runs all test categories
We can add more targets as we move forward.

There are two to-be-fixed-later in this PR:
1. Teach travis to run the testConnected tests. That requires travis to have a project and credentials. Added DR-114
1. Paths to databases are in both gradle.properties and application.properties and are hard to override. Added DR-115.

### CODE REVIEW

File: src/main/java/bio/terra/pdao/PrimaryDataAccess.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
+package bio.terra.pdao;
+
+import bio.terra.metadata.Study;
+
+/**
+ * In the long term, we want to make the primary data store be pluggable, supporting different implementations.
+ * This interface documents what that pluggable interface might look like. Of course, we never know until we
```
Comment (thathert) [inline src/main/java/bio/terra/pdao/PrimaryDataAccess.java:7]: There is a cliffhanger here :)

File: src/main/java/bio/terra/pdao/PrimaryDataAccess.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ * <ul>
+ *     <li>do sequence</li>
+ *     <ol>
+ *         <li>studyExists - validate that the study does not exist before calling create</li>
+ *         <li>createStudy - if it finds the study, it deletes it and recreates. That covers the case where the
+ *         create operation is interrupted by a service failure and restarts.</li>
+ *     </ol>
+ *     <li>undo</li>
+ *     <ol>
+ *     <li>deleteStudy - called on undo; does not fail if the study has already been deleted</li>
```
Comment (thathert) [inline src/main/java/bio/terra/pdao/PrimaryDataAccess.java]: nit: Cqn you indent this?

File: src/main/java/bio/terra/pdao/PrimaryDataAccess.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ *     </ol>
+ * </ul>
+ *
+ * REVIEWERS: Is it worth catching underlying implementation exceptions and packaging them into a PDAO exception?
+ * Seems of marginal value to me, but has a nice aesthetic.
```
Comment (thathert) [inline src/main/java/bio/terra/pdao/PrimaryDataAccess.java]: I like this pattern, prevents users of the PDAO from having to know too much about the inner workings of the PDAO.

File: src/main/java/bio/terra/pdao/bigquery/BigQueryConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+/**
+ * The theory of operation here is that each implementation of a pdao will specify a configuration.
+ * That way, implementation specifics can be separated from the interface. We'll see if it works out that way.
+ */
+@Configuration
+@Profile("bigquery")
+public class BigQueryConfiguration {
+
+    private static String rowIdColumnName = "datarepo_row_id";
+    private static String rowIdColumnDatatype = "STRING"; // TODO: datatype enums??
```
Comment (thathert) [inline src/main/java/bio/terra/pdao/bigquery/BigQueryConfiguration.java:19]: I added a ticket along with a doc for discussing "our" datatypes -> BQ datatypes.

https://broadinstitute.atlassian.net/jira/software/projects/DR/boards/517/backlog?selectedIssue=DR-118

**Review by thathert — CHANGES_REQUESTED** [review]
Looks good. I like the idea of adding a PDAO exception, if it's not too much work it'd be nice to have it in this PR.

File: src/main/java/bio/terra/pdao/bigquery/BigQueryConfiguration.java (hunk 4 above)
Comment (ddietterich) [reply src/main/java/bio/terra/pdao/bigquery/BigQueryConfiguration.java:19]: I had already added DR-112. I moved the contents of your description in DR-118 into DR-112, marked DR-118 as a duplicate and Done.

File: src/main/java/bio/terra/pdao/PrimaryDataAccess.java (hunk 1 above)
Comment (ddietterich) [reply src/main/java/bio/terra/pdao/PrimaryDataAccess.java:7]: I like to keep you on the edge of your seat...

File: src/main/java/bio/terra/pdao/PrimaryDataAccess.java (hunk 2 above)
Comment (ddietterich) [reply src/main/java/bio/terra/pdao/PrimaryDataAccess.java]: Done

File: src/main/java/bio/terra/pdao/PrimaryDataAccess.java (hunk 3 above)
Comment (ddietterich) [reply src/main/java/bio/terra/pdao/PrimaryDataAccess.java]: Implemented and committed

**Review by RoriCremer — APPROVED** [review]

**Review by thathert — APPROVED** [review]

**Pull request merged**